src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -120,9 +120,10 @@ class CollapsingNavigationMenu extends React.Component {
 
   componentDidUpdate() {
     const { isNewSelectedPath } = this.state;
-    const currentItemPosition = this.selectedItem && this.selectedItem.current ? this.selectedItem.current.getBoundingClientRect().bottom : null;
-    const navigationMenuHeight = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect().bottom;
-    if (isNewSelectedPath && currentItemPosition > navigationMenuHeight) {
+    const currentItemPosition = this.selectedItem && this.selectedItem.current ? this.selectedItem.current.getBoundingClientRect() : null;
+    const navigationMenuPosition = document.querySelector('#terra-dev-site-nav-menu').getBoundingClientRect();
+    if (isNewSelectedPath && currentItemPosition && navigationMenuPosition && (currentItemPosition.bottom > navigationMenuPosition.bottom
```

Comment:
can this condition be like this 
`if (isNewSelectedPath && (currentItemPosition.bottom > navigationMenuPosition.bottom
      || currentItemPosition.top < navigationMenuPosition.top))`  ? do we need  currentItemPosition and navigationMenuPosition here in this condition ?

Reply:
We need these, as the code may break if any of them is null.